asf_lifetime_cost_model/pipeline/cost_computations.py

```diff
@@ -0,0 +1,236 @@
+            "smallest", "no subsidy".
+        installation_year (int): The year in which the heating system is installed.
+
+    Raises:
+        ValueError: If the subsidy model is not supported.
+
+    Returns:
+        float: The subsidy value for the specified model.
+    """
+    supported_models = config.get("ashp_subsidy_options")
+
+    if subsidy_model not in supported_models:
+        raise ValueError(f"Unsupported subsidy model: {subsidy_model}. Supported models are: {supported_models}")
+
+    subsidy_data = data_getters.get_ashp_subsidy_options_data()
+    subsidy_value = subsidy_data[subsidy_data["model"] == subsidy_model][str(installation_year)].iloc[0]
+
+    return subsidy_value
+
+
+def compute_upfront_cost(
+    heating_system: str,
+    archetype: str,
+    annual_cost_reduction: float,
+    installation_year: int,
+    decile: int = None,
+    subsidy_model_or_value: Union[str, float] = 0,
+) -> float:
+    """Computes the upfront costs of installing a heating system.
+
+    Args:
+        heating_system (str): heating system.
+            Takes "ashp" (for air source heat pump) or "boiler" (for gas boiler).
+        archetype (str): Property archetype
+            Takes: TODO: add final list of archetypes.
+        decile (int, optional): cost decile, only used for air source heat pumps.
+        annual_cost_reduction (float): how much the cost of installing the heating system reduces each year
+            in comparison to the previous year.
+        installation_year (int): the year in which the heating system is installed.
+        subsidy_model_or_value (Union[str, float], optional): The subsidy model to get the subsidy to be subtracted
+            OR a fixed subsidy value.
+            Models include: "flat", "slow stepdown", "fast stepdown", "high", "zero from 2028",
+            "smallest", "no subsidy".
+
+    Raises:
+        ValueError: If the heating system inputed is not supported or the year is above or equal to 2035.
+
+    Returns:
+        float: Upfront cost of installing a heating system.
+    """
+    # Get installation cost for a specific heating system, archetype, and decile (where applicable)
+    installation_cost = get_installation_cost(heating_system=heating_system, archetype=archetype, decile=decile)
+
+    cost_reduction_data = creates_cost_reduction_data(annual_cost_reduction=annual_cost_reduction)
+    cost_reduction_value = cost_reduction_data[installation_year]
+
+    installation_cost = installation_cost * cost_reduction_value
+
+    if (type(subsidy_model_or_value) is str) and (heating_system == "ashp"):
+        subsidy_value = get_ashp_subsidy_value(
+            subsidy_model=subsidy_model_or_value, installation_year=installation_year
+        )
+        return installation_cost - subsidy_value
+    else:  # float provided
+        return installation_cost - subsidy_model_or_value
+
+
+def compute_total_lifetime_costs(
```

Comment:
this could obviously just take 3 parameters: maintenance, running and installation costs and sum them. But for now i'll leave it like this, as I don't know how we're organising these functions.